### DISCUSSION TITLE
How to enable video thumbnails?

### DISCUSSION THREAD

**Discussion by AlexValue** [Vanilla-OS/live-iso#366] [Q&A]
Apparently, in my previous distribution (Fedora), drivers for creating video thumbnails were installed automatically. Or they were installed from the software center (I don't remember exactly).

In VanillaOS, thumbnails do not work and I can't figure out what to install to fix this.
I asked ChatGPT for advice and he recommended installing the following drivers and rebooting the system:
`sudo apt install ffmpeg gstreamer1.0-libav gstreamer1.0-plugins-bad gstreamer1.0-plugins-ugly`

This advice did not help me and thumbnails are still not created. How to fix the problem?

**Comment by kbdharun** [comment]
Hi, we do miss some thumbnailers especially for video files, can you share the file formats which aren't showing a thumbnail in your device?

This would help me test and add it to our dev tag of desktop image (and it will be shipped in the next stable update to the image).

---

For now, you can try manually adding them using `abroot pkg add <package1> <package2> <...>` then `abroot pkg apply` and reboot. 

If this breaks the system, you can rollback to the previous state by pressing Shift during boot and selecting previous partition in GRUB bootloader, then proceed with the prompted rollback.

**Reply by AlexValue** [reply]
I have videos of different formats. Thumbnail generation does not work for any of them.

For convenience, I divided video formats into two groups. The first group is the videos that I have the most. The second group is the videos that I have, but are less important (from my point of view)

The first group:
```
.mp4
.webm
.mkv
.flv
.vob
.ogv
```

The second group:
```
.avi
.mov
.wmv
.divx
.ogg
.mpeg
```

Of the audio files, I only have `.mp3`, and thumbnail generation does not work for them. Perhaps there are other audio formats worth considering.

Thumbnail generation for images works correctly (.jpeg, .png, .webp, .gif, .svg, .avif)

Thumbnail generation for e-books works partially. Here is a list of my formats:

Works:
.epub

Doesn't work:
```
.fb2
.pdf
.djvu
```

**Reply by kbdharun** [reply]
Hi, thanks for the detailed response, I will look into the thumbnailer packages for these file formats and add them to the dev image soon.
